webapp/src/main/webapp/WEB-INF/jsp/portlet/spaceCreation.jsp

```diff
@@ -1,21 +1,56 @@
+<%@page import="org.exoplatform.services.security.ConversationState"%>
+<%@page import="org.exoplatform.portal.config.UserACL"%>
 <%@page import="org.exoplatform.container.ExoContainerContext"%>
 <%@page import="io.meeds.social.space.template.service.SpaceTemplateService"%>
 <%@page import="io.meeds.social.space.template.model.SpaceTemplate"%>
 <%@page import="io.meeds.social.space.template.model.SpaceTemplateFilter"%>
 <%@page import="io.meeds.social.util.JsonUtils"%>
 <%@page import="org.springframework.data.domain.Pageable"%>
 <%@page import="java.util.List"%>
+<%@page import="com.fasterxml.jackson.databind.ObjectMapper" %>
+<%@page import="java.util.Map" %>
+<%@page import="javax.portlet.PortletPreferences" %>
+<%@taglib uri="http://java.sun.com/portlet_2_0" prefix="portlet" %>
+<portlet:defineObjects />
+<portlet:actionURL var="saveSettingsUrl" />
 <%
+
+UserACL userAcl = ExoContainerContext.getService(UserACL.class);
+boolean isAdministrator = ConversationState.getCurrent() != null && userAcl.isAdministrator(ConversationState.getCurrent().getIdentity());
 SpaceTemplateService spaceTemplateService = ExoContainerContext.getService(SpaceTemplateService.class);
 SpaceTemplateFilter spaceTemplateFilter = new SpaceTemplateFilter(request.getRemoteUser(), request.getLocale(), false);
-List<SpaceTemplate> spaceTemplates = spaceTemplateService.getSpaceTemplates(spaceTemplateFilter, Pageable.unpaged(), true);
-%>
+List<SpaceTemplate> defaultSpaceTemplates = spaceTemplateService.getSpaceTemplates(spaceTemplateFilter, Pageable.unpaged(), true);
+String defaultJson = JsonUtils.toJsonString(defaultSpaceTemplates);
+
+Object rawSettings = request.getAttribute("settings");
+String settingsJson = null;
+if (rawSettings instanceof String[]) {
+  settingsJson = ((String[]) rawSettings)[0];
+}
+
+String settings = "{}";
+if (settingsJson != null) {
+  try {
+    ObjectMapper mapper = new ObjectMapper();
+    Map<String, Object> parsedSettings = mapper.readValue(settingsJson, Map.class);
+    settings = mapper.writeValueAsString(parsedSettings);
```

Comment:
Why transforming the Json String to Json String again ?

```suggestion
```

Reply:
Yes String content. How does it work in https://github.com/Meeds-io/social/blob/develop/webapp/src/main/webapp/WEB-INF/jsp/portlet/spacesList.jsp#L53

Isn't it the same ?